### PR TITLE
Task/07 api

### PR DESCRIPTION
**Pull request by roman-orlovskyi-moc**
**Acceptance Criteria for `1`**
- [x] Mentor is added to PR
- [x] AC copied to PR description
- [x] Link to deployed version is added to PR
- [x] `ProductPage` fetches data from the API using either `fetch` or `axios`
- [x] Pagination is implemented using query parameters
- [x] Search functionality is implemented using query parameters, with the search triggered by a search button
- [x] Category filtering is implemented using query parameters, resetting the page to the first page on change
- [x] A loading spinner or message is displayed while awaiting API requests
- [x] If the API response is an empty array or an error occurs, a "No products" message is displayed instead of a blank screen

**Acceptance Criteria for `1*`**
- [x] Pagination is hidden on mobile view
- [x] Infinite scroll is implemented for mobile view, either manually or using a library
- [x] Error notifications are implemented using `axios` interceptors, with component libraries for toastr notifications

[deployed version](https://roman-orlovskyi-moc.github.io/fe-react-2024/)

### CODE REVIEW

File: src/services/Api.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        axios.interceptors.response.use(
+            (response) => response,
+            (error) => {
+                toast.error(error.message);
+                return Promise.reject(error);
+            },
+        );
+    }
+
+    public static GetInstance(): ApiService {
```
Comment (vladyslav-yermolin-moc) [inline src/services/Api.service.ts]: Cool OOP solution with a static method.
Another option: you can export not the class itself, but the created example and use it as a singleton in whole app:

File: src/services/Api.service.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        try {
+            const response: AxiosResponse = await axios.get<T>(url, { params: parameters });
+
+            return response.data as T;
+        } catch (error) {
+            console.error('Error occurred while making GET request:', error);
+            throw error;
+        }
+    }
+}
```
Comment (vladyslav-yermolin-moc) [inline src/services/Api.service.ts:30]: ... Then you can use it in components without `GetInstance()`
```
export const axiosInstance = new ApiService();
```

**Review by vladyslav-yermolin-moc — APPROVED** [review]
Good job. If you like to implement my suggestion, please do it, if not, just merge as is

File: src/services/Api.service.ts (hunk 1 above)
Comment (roman-orlovskyi-moc) [reply src/services/Api.service.ts]: fixed

File: src/services/Api.service.ts (hunk 2 above)
Comment (roman-orlovskyi-moc) [reply src/services/Api.service.ts:30]: fixed

**Review by vladyslav-yermolin-moc — APPROVED** [review]

**Pull request merged**